File: leetcode-submissions/1514-minimum-value-to-get-positive-step-by-step-sum/solution.java
class Solution {
    public int minStartValue(int[] nums) {
        //prefix sums, if the smallest is < 1, return 1 - that
        // int[] prefixSums = new int[nums.length];
        int prevSum = nums[0];
        int smallestPrefix = prevSum;
        for(int i = 1; i < nums.length; i++) {
            prevSum += nums[i];
            if (prevSum < smallestPrefix) {
                smallestPrefix = prevSum;
            }
        }
        if (smallestPrefix < 1) {
            return 1 - smallestPrefix;
        }
        return 1; 
    }
}

File: leetcode-submissions/1514-minimum-value-to-get-positive-step-by-step-sum/solution.py
class Solution:
    def minStartValue(self, nums: List[int]) -> int:
        current_sum = 0
        lowest_sum = 0
        
        for numbers in nums:
            current_sum += numbers
            lowest_sum = min(current_sum, lowest_sum)
            
        ans = abs(lowest_sum) + 1
        return ans
